contributivity_measures.py

```diff
@@ -18,6 +18,48 @@
 
 #%% Compute independent performance scores of models trained independently on each node
 
+def compute_contributivity(method_to_compute, 
+                           current_scenario,
+                           sv_accuracy=0.01, 
+                           alpha=0.9, 
+                           contrib_accuracy=0.05):
+    
+    if method_to_compute == "Shapley values":
+        # Contributivity 1: Baseline contributivity measurement (Shapley Value)
+        (contributivity_scores, scores_var) = compute_SV(
+            current_scenario.node_list,
+            current_scenario.epoch_count,
+            current_scenario.x_val,
+            current_scenario.y_val,
+            current_scenario.x_test,
+            current_scenario.y_test,
+            current_scenario.aggregation_weighting,
+        )
+        score_dict = {"Shapley values": (contributivity_scores, scores_var)}
+    elif method_to_compute == "Independant scores":
+        # Contributivity 2: Performance scores of models trained independently on each node
+        scores = compute_independent_scores(
+            current_scenario.node_list,
+            current_scenario.epoch_count,
+            current_scenario.federated_test_score,
+            current_scenario.single_partner_test_mode,
+            current_scenario.x_test,
+            current_scenario.y_test,
+        )
+        score_dict = {"Independant scores raw": (scores[0], np.repeat(0.0, len(scores[0]))),
+                      "Independant scores additive": (scores[1], np.repeat(0.0, len(scores[1])))}
+    elif method_to_compute == "TMCS values":
+        # Contributivity 3: Truncated Monte Carlo Shapley
+        tmcs_results = truncated_MC(
+                        current_scenario, sv_accuracy, 
+                        alpha, contrib_accuracy
+                        )
+        score_dict = {"TMCS values": (tmcs_results["sv"], 
+                                      tmcs_results["std_sv"])}
+
+    
+    return score_dict
```

Comment:
Looks good! 
Naive question here: what if no `method_to_compute` is taken by the if/elif conditions (typo or other method name), `score_dict` is returned but not defined?

Reply:
Good point! I'll add a test when we initiate the scenario and I complete the conditional loop with a `else`.

Reply:
Yes good point to add test when we initiate the scenario so it will fail quickly !

Reply:
What you can do here is define `score_dict = {}` at the beginning of the method and remove this `else` condition. So, if a condition is met, `score_dict` is updated otherwise it remains empty and the method return the empty dict.

Reply:
Done

Reply:
@natct10 I think we should raise an exception here, if the `method_to_compute `is not amongst the predefined ones. I don't see the point of return an empty dict, since it will fail a bit after when calling `score_dict[score_method]` ?

Reply:
Good point!

Reply:
@RomainGoussault @natct10 But in fact this exception is raised during the initialization of scenario. 
- If a method is not in the default list or if there are no method in the list given in the config file, the script ends. 
- If there is no method field in config file, it consedered that it is the default situation, so it runs all methods.

Reply:
@celinejacques  Your second point is not right, you changed the code so it would rather exit the script ;)

Reply:
Okay, great!